app/controllers/api/internal/v1/laa_references_controller.rb

```diff
@@ -5,41 +5,56 @@ module Internal
     module V1
       class LaaReferencesController < ApplicationController
         def create
-          enforce_contract!
-          create_laa_reference
+          contract = NewLaaReferenceContract.new.call(**transformed_params)
+          enforce_contract!(contract)
+
+          LaaReferenceCreator.call(defendant_id: transformed_params[:defendant_id],
+                                   user_name: transformed_params[:user_name],
+                                   maat_reference: transformed_params[:maat_reference])
+
+          render status: :accepted
+        end
+
+        def destroy
+          contract = UnlinkDefendantContract.new.call(**transformed_params)
+          enforce_contract!(contract)
+
+          UnlinkLaaReferenceWorker.perform_async(
+            Current.request_id,
+            transformed_params[:defendant_id],
+            transformed_params[:user_name],
+            transformed_params[:unlink_reason_code],
+            transformed_params[:unlink_other_reason_text],
+          )
 
           render status: :accepted
         end
 
       private
 
-        def enforce_contract!
+        def enforce_contract!(contract)
           unless contract.success?
-            message = "LaaReference contract failed with: #{contract.errors.to_hash}"
+            message = "Contract failed with: #{contract.errors.to_hash}"
```

Comment:
Will contract.errors be verbose enough for us to know if it is the linking or unlinking that has failed? Or does sentry literally point to this file/line of code anyway?

Reply:
Good question. Sentry will tell us which controller action the error is happening on, so we should have enough information for debugging.

e.g. https://sentry.io/organizations/ministryofjustice/issues/2266587125 `Errors::ContractError Api::Internal::V1::LaaReferencesController#create`.